lib/src/components/fabs/fab.dart

```diff
@@ -113,58 +113,64 @@ class _ZetaFABState extends State<ZetaFAB> {
-                  horizontal: Zeta.of(context).spacing.large,
-                  vertical: Zeta.of(context).spacing.medium,
-                )
-              : EdgeInsets.all(widget.size.padding(context)),
-          child: Row(
-            mainAxisSize: MainAxisSize.min,
-            children: [
-              ZetaIcon(widget.icon, size: widget.size.iconSize(context)),
-              if (widget.expanded && widget.label != null)
-                Row(
-                  mainAxisSize: MainAxisSize.min,
-                  children: [Text(widget.label!, style: ZetaTextStyles.labelLarge)],
-                ),
-            ].divide(SizedBox(width: Zeta.of(context).spacing.small)).toList(),
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        FilledButton(
+          onPressed: widget.onPressed,
+          focusNode: widget.focusNode,
+          style: ButtonStyle(
+            padding: const WidgetStatePropertyAll(EdgeInsets.zero),
+            shape: WidgetStatePropertyAll(
+              widget.shape.buttonShape(isExpanded: widget.expanded, size: widget.size, context: context),
+            ),
+            backgroundColor: WidgetStateProperty.resolveWith((states) {
+              if (states.contains(WidgetState.disabled)) {
+                return Zeta.of(context).colors.surfaceDisabled;
+              }
+              if (states.contains(WidgetState.pressed)) {
+                return colors.selected;
+              }
+              if (states.contains(WidgetState.hovered)) {
+                return colors.hover;
+              }
+              return backgroundColor;
+            }),
+            side: WidgetStateProperty.resolveWith(
+              (Set<WidgetState> states) {
+                if (states.contains(WidgetState.focused)) {
+                  // TODO(UX-1134): This removes a defualt border when focused, rather than adding a second border when focused.
+                  return BorderSide(color: Zeta.of(context).colors.blue.shade50, width: ZetaBorders.medium);
+                }
+                return null;
+              },
+            ),
+          ),
+          child: AnimatedContainer(
+            duration: ZetaAnimationLength.normal,
+            child: Padding(
+              padding: widget.expanded
+                  ? EdgeInsets.symmetric(
+                      horizontal: Zeta.of(context).spacing.large,
+                      vertical: Zeta.of(context).spacing.medium,
+                    )
+                  : EdgeInsets.all(widget.size.padding(context)),
+              child: Row(
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  ZetaIcon(widget.icon, size: widget.size.iconSize(context)),
+                  if (widget.expanded && widget.label != null)
+                    Row(
+                      mainAxisSize: MainAxisSize.min,
+                      children: [Text(widget.label!, style: ZetaTextStyles.labelLarge)],
```

Comment:
This text should probably wrap if it gets too long, we don't want it too expand too far past the width of the button itself

Reply:
I've added a constraint of 100. I was looking for a decent way to set this dynamically but I fell short. Any thoughts?